lib/commons/dom/create-grid.js

```diff
@@ -254,65 +254,61 @@ function isFlexOrGridContainer(vNode) {
+ * @param {Number} index
  * @return {Number[]}
  */
-function createStackingOrder(vNode, parentVNode, nodeIndex) {
+function createStackingOrder(vNode, parentVNode, index) {
   const stackingOrder = parentVNode._stackingOrder.slice();
 
-  // if a positioned element has z-index: auto or 0 (step #8), or if
-  // a non-positioned floating element (step #5), treat it as its
-  // own stacking context
-  // @see https://www.w3.org/Style/css2-updates/css2/zindex.html
-  if (!isStackingContext(vNode, parentVNode)) {
-    if (vNode.getComputedStylePropertyValue('position') !== 'static') {
-      // Put positioned elements above floated elements
-      stackingOrder.push(createContext(POSITION_STATIC_ORDER, vNode));
-    } else if (vNode.getComputedStylePropertyValue('float') !== 'none') {
-      // Put floated elements above z-index: 0
-      // (step #5 floating get sorted below step #8 positioned)
-      stackingOrder.push(createContext(FLOAT_ORDER, vNode));
-    }
-    return stackingOrder;
-  }
-
   // if an element creates a stacking context, find the first
   // true stack (not a "fake" stack created from positioned or
   // floated elements without a z-index) and create a new stack at
   // that point (step #5 and step #8)
   // @see https://www.w3.org/Style/css2-updates/css2/zindex.html
-  const index = stackingOrder.findIndex(({ value }) =>
-    [ROOT_ORDER, FLOAT_ORDER, POSITION_STATIC_ORDER].includes(value)
-  );
-  if (index !== -1) {
-    stackingOrder.splice(index, stackingOrder.length - index);
+  if (isStackingContext(vNode, parentVNode)) {
+    const index = stackingOrder.findIndex(({ value }) =>
+      [ROOT_ORDER, FLOAT_ORDER, POSITION_ORDER].includes(value)
+    );
+    if (index !== -1) {
+      stackingOrder.splice(index, stackingOrder.length - index);
+    }
   }
 
   const zIndex = getRealZIndex(vNode, parentVNode);
   if (!['auto', '0'].includes(zIndex)) {
-    stackingOrder.push(createContext(parseInt(zIndex), vNode));
+    stackingOrder.push(createContext(parseInt(zIndex), vNode, index));
+    return stackingOrder;
+  }
+
+  // if a positioned element has z-index: auto or 0 (step #8), or if
+  // a non-positioned floating element (step #5), treat it as its
+  // own stacking context
+  // @see https://www.w3.org/Style/css2-updates/css2/zindex.html
+  if (vNode.getComputedStylePropertyValue('position') !== 'static') {
+    // Put positioned elements above floated elements
+    stackingOrder.push(createContext(POSITION_ORDER, vNode, index));
+    return stackingOrder;
+  }
+
+  if (vNode.getComputedStylePropertyValue('float') !== 'none') {
+    // Put floated elements above z-index: 0
+    // (step #5 floating get sorted below step #8 positioned)
+    stackingOrder.push(createContext(FLOAT_ORDER, vNode, index));
+    return stackingOrder;
+  }
+
+  if (!isStackingContext(vNode, parentVNode)) {
     return stackingOrder;
   }
-  // since many things can create a new stacking context without position or
```

Comment:
This is a nice change, I think including the `nodeIndex` in the comparator is much cleaner than the old way.